src/node_nanomsg.cc

```diff
@@ -191,35 +190,40 @@ NAN_METHOD(Err) {
   info.GetReturnValue().Set(Nan::New(nn_strerror(nn_errno())).ToLocalChecked());
 }
 
-typedef struct nanomsg_socket_s {
-  uv_poll_t poll_handle;
-  uv_os_sock_t sockfd;
-  Nan::Callback *callback;
-} nanomsg_socket_t;
+class NanoMsgPollCtx {
+  private:
+  public:
+    uv_poll_t poll_handle;
+    uv_os_sock_t sockfd;
+    Nan::Callback *callback;
+    NanoMsgPollCtx (int s, bool is_sender, Local<v8::Function> cb) {
+      callback = new Nan::Callback(cb);
+      size_t siz = sizeof(uv_os_sock_t);
+      nn_getsockopt(s, NN_SOL_SOCKET, is_sender ? NN_SNDFD : NN_RCVFD, &sockfd,
+          &siz);
+      poll_handle.data = this;
+    }
+    ~NanoMsgPollCtx () {
+      delete callback;
+    }
+};
 
-void NanomsgReadable(uv_poll_t *req, int status, int events) {
+static void NanomsgReadable(uv_poll_t *req, int status, int events) {
```

Comment:
I don't think so.  Any other compilation can forward declare `extern void NanomsgReadable ...` and the link against it (if `static` was removed).  Explicitly marking this function `static` restricts its linkage to this compilation unit.